include/proj/coordinateoperation.hpp

```diff
@@ -0,0 +1,366 @@
+/******************************************************************************
+ *
+ * Project:  PROJ
+ * Purpose:  ISO19111:2018 implementation
+ * Author:   Even Rouault <even dot rouault at spatialys dot com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2018, Even Rouault <even dot rouault at spatialys dot com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#ifndef COORDINATEOPERATION_HH_INCLUDED
+#define COORDINATEOPERATION_HH_INCLUDED
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "common.hpp"
+#include "io.hpp"
+#include "metadata.hpp"
+
+NS_PROJ_START
+
+namespace crs {
+class CRS;
+using CRSPtr = std::shared_ptr<CRS>;
+class DerivedCRS;
+class ProjectedCRS;
+}
+
+/** osgeo.proj.operation namespace
+
+  \brief Coordinate operations (relationship between any two coordinate
+  reference systems).
+*/
+namespace operation {
+
+// ---------------------------------------------------------------------------
+
+class CoordinateOperation : public common::ObjectUsage {
+  public:
+    PROJ_DLL virtual ~CoordinateOperation();
+    PROJ_DLL CoordinateOperation &
+    operator=(const CoordinateOperation &other) = delete;
+
+    PROJ_DLL const util::optional<std::string> &operationVersion() const;
+    PROJ_DLL const std::vector<metadata::PositionalAccuracy> &
+    coordinateOperationAccuracies() const;
+
+    // In the case of a derivingConversion of a DerivedCRS, sourceCRS() and
+    // targetCRS()
+    // will return null pointers if the owning DerivedCRS has been destroyed.
+    PROJ_DLL const crs::CRSPtr sourceCRS() const;
+    PROJ_DLL const crs::CRSPtr targetCRS() const;
+    PROJ_DLL const crs::CRSPtr &interplationCRS() const;
```

Comment:
Is this a spelling error? `interpOlation`?

Reply:
yes, fixed